fiftyone/server/samples.py

```
@@ -112,10 +112,15 @@ async def paginate_samples(
     )
-    try:
-        view = await run_sync_task(run, False)
-    except:
+    if after is None:
+        # first page, force dataset reload
         view = await run_sync_task(run, True)
+    else:
+        # not first page, optimistically skip dataset reload
+        try:
+            view = await run_sync_task(run, False)
+        except:
```

Ruff (0.11.9)

Do not use bare `except`

(E722)